src/xmipp/applications/tests/function_tests/test_metadata_db_main.cpp

```diff
@@ -61,17 +67,46 @@ class MetadataTest : public ::testing::Test
 
     // virtual void TearDown() {}//Destructor
 
-    MetaData mDsource,mDanotherSource;
-    MetaData mDunion, mDjoin;
+    MetaDataDb mDsource,mDanotherSource;
+    MetaDataDb mDunion, mDjoin;
     size_t id, id1,id2;
+    std::vector<int> mDsourceIds;
 };
 
+TEST_F( MetadataTest, IdIteration)
+{
+    auto it = mDsource.ids().begin();
+    for (size_t i = 0; i < mDsourceIds.size(); i++, ++it);
```

Comment:
add comment that the for loop is there to reach the end (`;` at the end is easy to miss)